include/strong_ptr/pointers.hpp

```diff
@@ -0,0 +1,1880 @@
+// Copyright 2024 - 2025 Khalil Estell and the libhal contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <array>
+#include <atomic>
+#include <memory_resource>
+#include <type_traits>
+#include <utility>
+
+using usize = std::uintptr_t;
+using i32 = std::int32_t;
+using u32 = std::uint32_t;
```

Comment:
We shouldn't polute the global namespace with our user defined types. Lets move this to the `mem` namespace.